### PR TITLE
Init Bridge Transfers: Add Hop & OP Standard Bridge

### PR DESCRIPTION
**Pull request by MSilb7**
Brief comments on the purpose of your changes:
**Opening a draft PR for visibility & in-progress checks/feedback:**
1. Create bridges.bridge_transfers (DONE)
2. Add Hop Protocol Support for OP (DONE)
3. Add The Optimism Standard Bridge (DONE)

Future PR:
Extend Hop Protocol Support for other chains (just letting OP go through for now to keep it simpler, but it's set up to be parameterized).
- This is slightly complicated since some chains have multiple hop schemas ("hop" and "hop_protocol")

I will make this ready for review once the updates are all in.

**For Dune Engine V2**

I've checked that:

### General checks:
* [ ] I tested the query on dune.com after compiling the model with dbt compile (compiled queries are written to the target directory)
* [ ] I used "refs" to reference other models in this repo and "sources" to reference raw or decoded tables 
* [ ] if adding a new model, I added a test
* [ ] the filename is unique and ends with .sql
* [ ] each sql file is a select statement and has only one view, table or function defined  
* [ ] column names are `lowercase_snake_cased`
* [ ] if adding a new model, I edited the dbt project YAML file with new directory path for both models and seeds (if applicable)
* [ ] if wanting to expose a model in the UI (Dune data explorer), I added a post-hook in the JINJA config to add metadata (blockchains, sector/project, name and contributor Dune usernames)

### Pricing checks:
* [ ] `coin_id` represents the ID of the coin on coinpaprika.com
* [ ] all the coins are active on coinpaprika.com (please remove inactive ones)

### Join logic:
* [ ] if joining to base table (i.e. ethereum transactions or traces), I looked to make it an inner join if possible

### Incremental logic:
* [ ] I used is_incremental & not is_incremental jinja block filters on both base tables and decoded tables
  * [ ] where block_time >= date_trunc("day", now() - interval '1 week')
* [ ] if joining to base table (i.e. ethereum transactions or traces), I applied join condition where block_time >= date_trunc("day", now() - interval '1 week')
* [ ] if joining to prices view, I applied join condition where minute >= date_trunc("day", now() - interval '1 week')

### CODE REVIEW

**Comment by MSilb7** [comment]
Same issue as https://github.com/duneanalytics/spellbook/pull/2042 where table variables are causing a check to fail:

Check the script has existing refs and sources...........................Failed
- hook id: check-script-ref-and-source
- exit code: 1

Missing model (ref) bridge_protocol_model
Missing model (ref) hop_tf_model

**Comment by MSilb7** [comment]
Struggling to get `ovm_optimism` to work in the dbt_project.yml, but otherwise should be good to go

**Comment by MSilb7** [comment]
Woohoo, test passed

File: models/optimism/bridge/ovm_optimism_standard_bridge_transfers.sql
Comment (soispoke) [inline models/optimism/bridge/ovm_optimism_standard_bridge_transfers.sql:1]: General comment on this model:

I think it's a good idea to include columns we haven't necessarily considered in our bridge.flows spell, namely the source/destination chain id/name, the `fee_amount`, and the `is_native_bridge` field.
For the other ones, I think it would make sense to change their names so they can follow naming conventions that can be found in this PR: https://github.com/duneanalytics/spellbook/pull/2212

wdyt?

File: models/hop_protocol/optimism/hop_protocol_optimism_bridge_transfers.sql
Comment (soispoke) [inline models/hop_protocol/optimism/hop_protocol_optimism_bridge_transfers.sql:3]: If we follow conventions from the other PR, this model would be named `hop_protocol_optimism_flows.sql`, I don't have strong opinions on this particular name (could also be `hop_protocol_optimism_bridge_flows.sql`) but we should standardize that

File: models/optimism/bridge/ovm_optimism_standard_bridge_transfers.sql
Comment (soispoke) [inline models/optimism/bridge/ovm_optimism_standard_bridge_transfers.sql:28]: 'deposit' should be lowercase

File: models/optimism/bridge/ovm_optimism_standard_bridge_transfers.sql
Comment (soispoke) [inline models/optimism/bridge/ovm_optimism_standard_bridge_transfers.sql:63]: 'withdrawal' should be lowercase

File: models/optimism/bridge/ovm_optimism_standard_bridge_transfers.sql
Comment (soispoke) [inline models/optimism/bridge/ovm_optimism_standard_bridge_transfers.sql:119]: We should make sure to ensure types by explicitly casting amount using CAST(`x` AS DOUBLE) for example

File: models/bridges/readme.md
Comment (soispoke) [inline models/bridges/readme.md:1]: Super useful readme, thanks!

File: models/bridges/bridges_bridge_transfers.sql
Comment (soispoke) [inline models/bridges/bridges_bridge_transfers.sql:1]: Please look at this PR to deal with meta tables like nft.trades, bridge.flows, etc...
https://github.com/duneanalytics/spellbook/pull/2212

I don"t think we should duplicate names (bridges_bridge_transfers) in this case, but it's a matter of naming conventions (imo bridge.flows is better than bridges_bridge_transfers here)

**Review by soispoke — COMMENTED** [review]
Awesome work thanks @MSilb7 

Looking forward to work on this and set up the bridges spells so other wizards can build on your efforts 🔥

File: models/optimism/bridge/ovm_optimism_standard_bridge_transfers.sql
Comment (MSilb7) [reply models/optimism/bridge/ovm_optimism_standard_bridge_transfers.sql:119]: on this you mean the `...token_amount_raw`? Or decimals as well?

File: models/hop_protocol/optimism/hop_protocol_optimism_bridge_transfers.sql
Comment (MSilb7) [reply models/hop_protocol/optimism/hop_protocol_optimism_bridge_transfers.sql:3]: yeah I'm good standardizing to `flows`

File: models/optimism/bridge/ovm_optimism_standard_bridge_transfers.sql
Comment (MSilb7) [reply models/optimism/bridge/ovm_optimism_standard_bridge_transfers.sql:1]: Yeah, I'll look through and try to standardize with `flows`

`fee_amount` I'm still unsure about, I just included it for now since Hop made it easy to tease out.

On `is_native_bridge` the real meta question is how we avoid duplication between bridge protocols, standard bridges, and each end of the bridge event (i.e. deposit from chain A perspective, withdrawal on Chain B perspective - We wouldn't sum both as "volume").

Comment (soispoke) [reply models/optimism/bridge/ovm_optimism_standard_bridge_transfers.sql:1]: for `fee_amount` I suggest we keep it in the `hop` models but don't include it in the meta `bridge.flows` model because I don't think it's always easy/obvious how to get fees for the different kinds of bridges.

Comment (soispoke) [reply models/optimism/bridge/ovm_optimism_standard_bridge_transfers.sql:1]: I'll add source and destination chain id/name in the Celer PR as well #2212

File: models/bridges/bridges_bridge_transfers.sql
Comment (soispoke) [inline models/bridges/bridges_bridge_transfers.sql:34]: lowercase for deposit and withdrawal

File: dbt_project.yml
Comment (augustog) [inline dbt_project.yml:107]: Is this removal desired?

**Comment by soispoke** [comment]
Changes due to schema change for decoded `hop_protocol` tables done

File: models/bridge/bridges_schema.yml
Comment (augustog) [inline models/bridge/bridges_schema.yml:71]: Good point to capture fees, and from what I see here, it's possible to have fees paid in different asset than the bridged token, correct? (this would make sense).

I would probably adjust the naming a bit, since `token_fee_...` seems a bit confusing to me.

Í'd go:

<html>
<body>
<b style="font-weight:normal;" id="docs-internal-guid-7c602076-7fff-35cd-4d3e-eb591087acd2"><div dir="ltr" style="margin-left:0pt;" align="left">

Column name | Column description | dtype | Soft / Hard requirements* | Example
-- | -- | -- | -- | --
fee_amount | fee amount paid denominated in the asset used | numeric | not null | 21.2
fee_amount_usd | fee amount denominated in USD | numeric | not null | 22.7
fee_amount_raw | ... | numeric | not null | 21233124100014230
fee_token_symbol | asset used to pay the bridge fee | string | not null | USDC
fee_token_address | contract address of the asset used to pay the fee | string | not null | 0x..

</div></b>
</body>
</html>

File: models/bridge/readme.md
Comment (augustog) [inline models/bridge/readme.md:1]: absolutely love this

we should make it into a doc block and add it to the model description (we can do that after merging tho)

File: models/chain_info/chain_ids.sql
Comment (augustog) [inline models/chain_info/chain_ids.sql:5]: this sounds like metadata we should mostly use in spells for now (I'd like us to structure a bit better how we expose these kinds of datasets, also e.g. ABI & contracts), do we want to expose this in the UI now?

File: models/chain_info/chain_info_schema.yml
Comment (augustog) [inline models/chain_info/chain_info_schema.yml:12]: smol typo

File: models/chain_info/chain_ids.sql
Comment (augustog) [reply models/chain_info/chain_ids.sql:5]: I don't think it's a major issue if these kinds of tables stay inside e.g. a project schema, but if we create a new sector for `chain_info` I'd prefer to have a more structured approach to how we present this on the UI (without blocking this PR for it, hence I propose let's just hide it)

File: models/hop_protocol/hop_protocol_schema.yml
Comment (augustog) [inline models/hop_protocol/hop_protocol_schema.yml:63]: Same naming comment as above

**Review by augustog — CHANGES_REQUESTED** [review]
Just had a pass and it looks perfect, I have only a small handful of inline comments mostly to do with naming conventions and what we show in the UI.
- I suggested a rename for the bridge fee columns, since token_fee_* seemed a bit counterintuitive.
- The chain_info static table feels like it shouldn't live on the same level as sector / project spells, I suggest we probably hide it from the UI and think of where we wanna have it (woulnd't block this PR with that discussion)

Other than that my only comment would be around testing, we should probably have a seed-type test and a handful of sample transactions. I'd however merge as-is and start validating it live, and add this as a next step.

File: dbt_project.yml
Comment (MSilb7) [reply dbt_project.yml:107]: oh, no that must have been some merge bug, fixing it.

File: models/bridge/readme.md
Comment (MSilb7) [reply models/bridge/readme.md:1]: Yeah, open to see what that looks like after (I'm still a dbt noob)

File: models/chain_info/chain_ids.sql
Comment (MSilb7) [reply models/chain_info/chain_ids.sql:5]: Yeah, we can hide it for now. Some bridges use a different chain id structure as well, so there could be multiple chain id formats to consider later on.

**Comment by MSilb7** [comment]
Pushed updates re: @augustog 's comments

**Review by augustog — APPROVED** [review]

**Comment by augustog** [comment]
On the tests breaking:
- The pre commit hook is failing due to a known issue ✅ 
- Seems like `hop_protocol_optimism_flows.sql` has a broken dep? ⚠️  
    - `Table or view not found: hop_protocol_gnosis.L2_xDaiBridge_evt_TransferSent` (seems we don't have hop contracts decoded for Gnosis?!)

**Comment by MSilb7** [comment]
Oh wait actually it does look like the contracts went away.

`SELECT l2Bridge FROM hop_protocol.addresses WHERE blockchain = 'gnosis'` :)

0x25d8039bb044dc227f741a9e381ca4ceae2e6ae8
0xfd5a186a7e8453eb867a360526c5d987a00acac2
0x7ac71c29fedf94bac5a5c9ab76e1dd12ea885ccc
0x0460352b91d7cf42b0e1c1c30f06b602d9ef2238
0xd8926c12c0b2e5cd40cfda49ecaff40252af491b
0x6f03052743cd99ce1b29265e377e320cd24eb632

**Comment by 0xRobin** [comment]
@MSilb7 FYI, you can push out the `ref()` to the variable definition to make the pre-commit hook happy.
```sql
{% set models = [
 ref('model1')
,ref('model2')
] %}


{% for ref_model in models %}
SELECT *
FROM {{ ref_model }}
...
```

**Comment by MSilb7** [comment]
> @MSilb7 FYI, you can push out the `ref()` to the variable definition to make the pre-commit hook happy.
> 
> ```sql
> {% set models = [
>  ref('model1')
> ,ref('model2')
> ] %}
> 
> 
> {% for ref_model in models %}
> SELECT *
> FROM {{ ref_model }}
> ...
> ```

nice, tried an update

**Comment by soispoke** [comment]
tagging @antonio-mendes for visibility on decoded gnosis contracts

**Comment by soispoke** [comment]
@MSilb7 @augustog We're rebuilding the decoded tables under the `hop_protocol` schema for consistency, will update you when it's all done

**Comment by antonio-mendes** [comment]
gm everyone!

Update from the decoding side:
-> contracts @MSilb7 are indeed decoded properly: https://dune.com/queries/1746380
-> view generation was stuck which is why you weren't seeing them on the ui, should be showing up now/soon
-> saw 1 contract had the wrong name, and processing the rename now:
image

🖖

**Comment by MSilb7** [comment]
> gm everyone!
> 
> Update from the decoding side: -> contracts @MSilb7 are indeed decoded properly: https://dune.com/queries/1746380 -> view generation was stuck which is why you weren't seeing them on the ui, should be showing up now/soon -> saw 1 contract had the wrong name, and processing the rename now: image
> 
> 🖖

Awesome! I see there's still some errors on my end from old schemas, updating now.

File: models/optimism/standard_bridge/optimism_standard_bridge_schema.yml
Comment (soispoke) [reply models/optimism/standard_bridge/optimism_standard_bridge_schema.yml:51]: @augustog About the Celer PR, the goal is to merge this one first and then I'll adapt https://github.com/duneanalytics/spellbook/pull/2212 so it matches changes/updates we've made to Hop+Standard bridge

**Comment by soispoke** [comment]
everything lgtm

**Pull request merged**
